server/src/main/java/com/saralein/server/router/Router.java

```diff
@@ -15,17 +15,20 @@
     private final Path root;
     private final Controller directoryController;
     private final Controller fileController;
+    private final Controller partialContentController;
     private final ErrorController errorController;
     private boolean resourceExists = false;
     private boolean resourceIsDirectory = false;
```

Comment:
So, unrelated to the specific feature, but I was thinking about these resourceExists/resourceIsDirectory properties. They don't seem to be router state, but rather request state. Can we refactor these away from being instance properties and evaluate and use them when they're needed?

Reply:
Yeah, I don't think these should be here.  Trying to rework it a bit on [this PR](https://github.com/saralein/java-server/pull/27).